joomla/joomla-cms
# Template replace

### Description

This PR adds `regexReplaceInTemplate` function to JDocumentHTML

This can be used for adding/replacing html in the current template, like, for instance, adding HTML5 data-\* attributes used by javascript.

Examples of usage:

```
JFactory::getDocument()->regexReplaceInTemplate('<body', '<body data-json=\''.json_encode(array('item1' => 'value1', 'item2' => 'value2')).'\'');
JFactory::getDocument()->regexReplaceInTemplate('<body', '<body data-var="1"');
```

Of course, since is a regex replace of the template HTML this can be used for many other things before the final html is rendered.
### How to test
1. Add one of the examples above somewhere in /libraries/cms/html/behaviour.php. I added in the `caption` function.
2. Go to homepage and check if in the HTML code there is the following code:
   `<body data-json='{"item1":"value1","item2":"value2"}'` 
### B/C

None
